public/pages/utils/anomalyResultUtils.ts

```diff
@@ -167,9 +175,272 @@ export const filterWithDateRange = (
   dateRange: DateRange,
   timeField: string
 ) => {
-  const anomalies = data.filter(item => {
-    const time = get(item, `${timeField}`);
-    return time && time >= dateRange.startDate && time <= dateRange.endDate;
+  const anomalies = data
+    ? data.filter(item => {
+        const time = get(item, `${timeField}`);
+        return time && time >= dateRange.startDate && time <= dateRange.endDate;
+      })
+    : [];
+  return anomalies;
+};
+
+export const RETURNED_AD_RESULT_FIELDS = [
```

Comment:
nit: maybe you can use AD_DOC_FIELDS instead: https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/master/server/utils/constants.ts#L47

Reply:
Thanks. We define less fields here. This is to avoid retrieving extra fields to improve performance.